framework/logging.cpp

```diff
@@ -1624,8 +1621,7 @@ inline AbstractLogger::AbstractLogger(const struct test *test, ChildExitStatus s
         PerThreadData::Common *data = sApp->thread_data(i);
         ThreadState thr_state = data->thread_state.load(std::memory_order_relaxed);
         if (data->has_failed()) {
-            if (data->fail_time != 0 && data->fail_time < earliest_fail)
-                earliest_fail = data->fail_time;
+            earliest_fail = std::min(earliest_fail, data->fail_time);
```

Comment:
It uses the fact that "invalid_time_point" is `MonotonicTimePoint::max()`. Can we define a constant for "invalid_time_point" and static_assert that our assumption holds?
This might be painful to debug later if we ever change the invalid time point value.
If we `static_assert` it here we explicitly state the assumption made.

Reply:
I can define a constant, but I don't see a way to `static_assert` it.